dockers/docker-snmp/snmp_yml_to_configdb.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+
+import yaml
+from swsssdk import ConfigDBConnector
+
+db = ConfigDBConnector()
+db.connect()
+
+with open('/etc/sonic/snmp.yml', 'r') as yaml_file:
+    yaml_snmp_info = yaml.load(yaml_file)
+
+full_snmp_comm_list = ['snmp_rocommunity', 'snmp_rocommunities', 'snmp_rwcommunity', 'snmp_rwcommunities']
+
+for comm_type in full_snmp_comm_list:
+    if comm_type in yaml_snmp_info.keys():
+        if comm_type.startswith('snmp_rocommunities'):
+            for community in yaml_snmp_info[comm_type]:
+                db.set_entry('SNMP_COMMUNITY', community, {"TYPE": "RO"})
```

Comment:
>set_entry [](start = 19, length = 9)

If snmp.yml conflicts with ConfigDB, you overwrite. Should we treat ConfigDB as preferred?

Reply:
So the plan is to port everything over to ConfigDB but we haven't updated the snmpd.conf.j2 (jinja) template yet.  So even though we are moving everything from snmp.yml over to ConfigDB we aren't using ConfigDB to update the snmpd.conf file that is generated from the snmpd.conf.j2 jinja template.  This way there is no conflict.  

The next PR I submit will only generate the snmpd.conf file from the ConfigDB (as I will change the snmpd.conf.j2 jinja template to only use that).  This way there will be no conflict as the snmp.yml file info will be dumped into ConfigDB and after that the snmpd.conf file will be generated from ConfigDB only.  

I did this in two different PR's to make things easier.  This one is non-instrusive.  The next one will allow us to seemlessly migrate from snmp.yml to configdb (via the jinja template).

Let me know if this answers you questions.  There should be no way to have conflicts doing it this way.

Reply:
Added code to ensure the ConfigDB is always the source of truth and the failback will be the snmp.yml file if it exists but is not in the configdb. 

So currently when a device is first deployed it will only have the snmp.yml file and nothing in configdb.  The snmp.yml file info will be moved in the configdb.  That means as long as the configdb is saved the next reload the info from the configdb will be the source of truth. 

Here is the testing:

```
admin@str-s6000-acs-11:~$ redis-cli -n 4 keys "*" | grep -i snmp 
SNMP_COMMUNITY|public
FEATURE|snmp
SNMP|CONTACT
SNMP|LOCATION
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|CONTACT"
1) "joe"
2) "joe@contoso.com"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
1) "Location"
2) "Emerald City"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|public"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ cat /etc/sonic/snmp.yml
snmp_location: Mountains
snmp_rocommunities: [Kat]
snmp_rwcommunities: [Joker, Travis]
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ docker stop snmp
snmp
admin@str-s6000-acs-11:~$ docker start snmp
snmp
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ redis-cli -n 4 keys "*" | grep -i snmp 
SNMP_COMMUNITY|Joker
SNMP_COMMUNITY|public
FEATURE|snmp
SNMP_COMMUNITY|Kat
SNMP|CONTACT
SNMP|LOCATION
SNMP_COMMUNITY|Travis
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|public"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|CONTACT"
1) "joe"
2) "joe@contoso.com"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
1) "Location"
2) "Emerald City"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Joker"
1) "TYPE"
2) "RW"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Kat"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ cat /etc/sonic/snmp.yml
snmp_location: Mountains
snmp_rocommunities: [Kat]
snmp_rwcommunities: [Joker, Travis]
admin@str-s6000-acs-11:~$ 
```

Reply:
Also checked to confirm that if "public" is RO in configdb but I update snmp.yml to have "public" as RW it will use what is in configdb.

```
admin@str-s6000-acs-11:~$ cat /etc/sonic/snmp.yml
snmp_location: Mountains
snmp_rocommunities: [Kat]
snmp_rwcommunities: [Joker, Travis, public]
admin@str-s6000-acs-11:~$ redis-cli -n 4 keys "*" | grep -i snmp 
SNMP_COMMUNITY|Joker
SNMP_COMMUNITY|public
FEATURE|snmp
SNMP_COMMUNITY|Kat
SNMP|CONTACT
SNMP|LOCATION
SNMP_COMMUNITY|Travis
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Joker"
1) "TYPE"
2) "RW"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Kat"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Travis"
1) "TYPE"
2) "RW"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|public"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
1) "Location"
2) "Emerald City"
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ docker stop snmp
snmp
admin@str-s6000-acs-11:~$ docker start snmp
snmp
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ 
admin@str-s6000-acs-11:~$ redis-cli -n 4 keys "*" | grep -i snmp 
SNMP_COMMUNITY|Joker
SNMP_COMMUNITY|public
FEATURE|snmp
SNMP_COMMUNITY|Kat
SNMP|CONTACT
SNMP|LOCATION
SNMP_COMMUNITY|Travis
admin@str-s6000-acs-11:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|public"
1) "TYPE"
2) "RO"
admin@str-s6000-acs-11:~$ 
```